geode-core/src/distributedTest/java/org/apache/geode/internal/cache/RebalanceWhileCreatingRegionDistributedTest.java

```diff
@@ -0,0 +1,298 @@
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion done wait for Before_RemoveBucketMessage gate");
+    createRegion(regionName, RegionShortcut.PARTITION_PROXY);
+    logger.info(
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion about to signal After_CreateProxyRegion gate");
+    blackboard.signalGate(AFTER_CREATE_PROXY_REGION);
+    logger.info(
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion done signal After_CreateProxyRegion gate");
+  }
+
+  private void waitToCreateSingleBucketProxyRegion(String regionName) throws Exception {
+    logger.info(
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion about to wait for Before_RemoveBucketMessage gate");
+    blackboard.waitForGate(BEFORE_REMOVE_BUCKET_MESSAGE);
+    logger.info(
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion done wait for Before_RemoveBucketMessage gate");
+    createSingleBucketRegion(regionName, RegionShortcut.PARTITION_PROXY);
+    logger.info(
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion about to signal After_CreateProxyRegion gate");
+    blackboard.signalGate(AFTER_CREATE_PROXY_REGION);
+    logger.info(
+        "RebalanceWhileCreatingRegionDistributedTest.waitToCreateRegion done signal After_CreateProxyRegion gate");
+  }
+
+  private void addDistributionMessageObserver(String regionName) {
+    DistributionMessageObserver.setInstance(new TestDistributionMessageObserver(regionName));
+  }
+
+  class TestDistributionMessageObserver extends DistributionMessageObserver {
+
+    private final String regionName;
+
+    public TestDistributionMessageObserver(String regionName) {
+      this.regionName = regionName;
+    }
+
+    public void beforeProcessMessage(ClusterDistributionManager dm, DistributionMessage message) {
+      if (message instanceof RemoveBucketMessage) {
+        logger.info(
+            "TestDistributionMessageObserver.beforeProcessMessage about to signal Before_RemoveBucketMessage gate");
+        blackboard.signalGate(BEFORE_REMOVE_BUCKET_MESSAGE);
+        logger.info(
+            "TestDistributionMessageObserver.beforeProcessMessage done signal Before_RemoveBucketMessage gate");
+      }
+    }
+
+    public void beforeSendMessage(ClusterDistributionManager dm, DistributionMessage message) {
+      if (message instanceof DestroyRegionOperation.DestroyRegionMessage) {
+        DestroyRegionOperation.DestroyRegionMessage drm =
+            (DestroyRegionOperation.DestroyRegionMessage) message;
+        if (drm.regionPath.contains(regionName)) {
+          logger.info(
+              "TestDistributionMessageObserver.beforeSendMessage about to wait for After_CreateProxyRegion gate regionName={}",
+              drm.regionPath);
+          try {
+            blackboard.waitForGate(AFTER_CREATE_PROXY_REGION);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
```

Comment:
In DUnit tests, you can usually remove the try-catch blocks and just keep adding throws checked exceptions to every method.

Reply:
This method overrides `DistributionMessageObserver.beforeSendMessage()` which does not have throws declaration.